subsys/bluetooth/controller/ll_sw/ull_master.c

```diff
@@ -553,6 +578,46 @@ uint8_t ll_enc_req_send(uint16_t handle, uint8_t const *const rand,
 }
 #endif /* CONFIG_BT_CTLR_LE_ENC */
 
+void ull_master_cleanup(struct node_rx_hdr *rx_free)
+{
+	struct node_rx_ftr *ftr = &rx_free->rx_ftr;
+	struct ll_scan_set *scan =
+		(void *)HDR_LLL2EVT(ftr->param);
+	struct lll_conn *conn_lll;
+	struct ll_conn *conn;
+	memq_link_t *link;
+
+	conn_lll = scan->lll.conn;
+	LL_ASSERT(conn_lll);
+	scan->lll.conn = NULL;
+
+	LL_ASSERT(!conn_lll->link_tx_free);
+	link = memq_deinit(&conn_lll->memq_tx.head,
+			   &conn_lll->memq_tx.tail);
+	LL_ASSERT(link);
+	conn_lll->link_tx_free = link;
+
+	conn = (void *)HDR_LLL2EVT(conn_lll);
+	ll_conn_release(conn);
+
+	scan->is_enabled = 0U;
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT) && defined(CONFIG_BT_CTLR_PHY_CODED)
+	scan->lll.phy = 0U;
+
+	struct ll_scan_set *scan_coded =
+				ull_scan_is_enabled_get(SCAN_HANDLE_PHY_CODED);
+	if (scan_coded && scan_coded != scan) {
+		conn_lll = scan_coded->lll.conn;
+		LL_ASSERT(conn_lll);
+		scan_coded->lll.conn = NULL;
+
```

Comment:
Why is there no memq_deinit() and ll_conn_release() for the connection in the Coded PHY scanner? If it is supposed to be the same as for the 1M PHY scanner, then a comment and/or assert would be good.

Reply:
I will add comments to explain how the single connection context is release considering the 3 cases: 
1. Initiating only on 1M
2. Initiating only on Coded PHY
3. Initiating on both 1M and Coded PHY